src/rtpghi~.c

```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
     else {
         post("failed to init, status %d", init_s);
     }
+   
+   x->c = getbytes(M * sizeof *(x->c));
+   post("length of c[]: %d", (M/2+1) * sizeof *(x->c));
+   post("s_n: %d", sp[0]->s_n);
+   
+   dsp_add(rtpghi_tilde_perform, 3,x, sp[0]->s_vec, sp[1]->s_vec, sp[2]->s_vec, sp[0]->s_n);
+}
+
+void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
+{
+
+  t_rtpghi_tilde *x = (t_rtpghi_tilde *)pd_new(rtpghi_tilde_class);
+
+    x->ol_pd=atom_getfloat(argv);
+    x->M_pd=atom_getfloat(argv+1);
+    x->tol_pd=atom_getfloat(argv+2);
+    x->do_causal_pd=atom_getfloat(argv+3);
+    x->window_type= *atom_getsymbol(argv+4);
+    x->sta_pd=NULL;
+    x-> c = NULL;
     
-    outlet_new(&x->x_obj, gensym("signal"));
-    outlet_new(&x->x_obj, gensym("signal"));
+    outlet_new(&x->x_obj, &s_signal);
+    outlet_new(&x->x_obj, &s_signal);
     
-    x->out = (t_sample **)getbytes(2 * sizeof(t_sample *));
-    x->c = (ltfat_complex_s *)getbytes(M * (sizeof *(x->c)));
-    x->out[0] = 0;
-    x->out[1] = 0;
-        
+    x->out_real = getbytes(sizeof (t_sample *));
+	x->out_imag = getbytes(sizeof (t_sample *));
+	
   return (void *)x;
 }
 
 
 void rtpghi_tilde_free(t_rtpghi_tilde *x)
 {
-    ltfat_int M = x->M_pd;
-    post("destroyed state at adrr: %p\n", &(x->sta_pd));
-    phaseret_rtpghi_done_s(&(x->sta_pd));
-    post("x->M_pd: %d", M);
-    freebytes(x->c, (M) * sizeof *(x->c));
-    freebytes(x->out,2 * sizeof(t_sample *));
+	ltfat_int M = x->M_pd;
+	post("destroyed state at adrr: %p\n", &(x->sta_pd));
+	phaseret_rtpghi_done_s(&(x->sta_pd));
+
+	freebytes(x->c, (M/2+1) * sizeof *(x->c));
+	freebytes(x->out_real, sizeof (t_sample*));
+	freebytes(x->out_imag, sizeof (t_sample *));
 }
 
 void rtpghi_tilde_setup(void) {
 
-    rtpghi_tilde_class = class_new(gensym("rtpghi~"),
-                               (t_newmethod)rtpghi_tilde_new,
-                                (t_method)rtpghi_tilde_free,
-                               sizeof(t_rtpghi_tilde),
-                               CLASS_DEFAULT,
-                                A_GIMME,
-                               0);
-    class_addmethod(rtpghi_tilde_class,
-                    (t_method)rtpghi_tilde_dsp, gensym("dsp"), A_CANT, 0);
-    CLASS_MAINSIGNALIN(rtpghi_tilde_class, t_rtpghi_tilde, f);
+   rtpghi_tilde_class = class_new(gensym("rtpghi~"),
+                              (t_newmethod)rtpghi_tilde_new,
+                               (t_method)rtpghi_tilde_free,
+                              sizeof(t_rtpghi_tilde),
+                              CLASS_DEFAULT,
+                               A_GIMME,
+                              0);
+   class_addmethod(rtpghi_tilde_class,
+                   (t_method)rtpghi_tilde_dsp, gensym("dsp"), A_CANT, 0);
+   CLASS_MAINSIGNALIN(rtpghi_tilde_class, t_rtpghi_tilde, f);
```

Comment:
also add `class_addmethod(rtpghi_tilde_class, (t_method)rtpghi_tilde_causal, gensym("causal"), A_FLOAT, 0);` and friends.